From 58c8ad7ee812916eb59f7f93791909eea383dc21 Mon Sep 17 00:00:00 2001
From: Ben Greear <greearb@candelatech.com>
Date: Tue, 29 Jan 2013 22:48:28 -0800
Subject: [PATCH 13/37] nfs: Allow up to 5000 NFSv4 connections (instead of
 1024).

Signed-off-by: Ben Greear <greearb@candelatech.com>
---
 fs/nfs/callback.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/nfs/callback.c b/fs/nfs/callback.c
index cd9d992..acdd360 100644
--- a/fs/nfs/callback.c
+++ b/fs/nfs/callback.c
@@ -298,7 +298,7 @@ static struct svc_serv *nfs_callback_create_svc(int minorversion)
 	/* As there is only one thread we need to over-ride the
 	 * default maximum of 80 connections
 	 */
-	serv->sv_maxconn = 1024;
+	serv->sv_maxconn = 5000;
 	dprintk("nfs_callback_create_svc: service created\n");
 	return serv;
 }
-- 
2.4.11

