From ccaba623485be184c83af781f61ba8e20e2e325a Mon Sep 17 00:00:00 2001
From: Ben Greear <greearb@candelatech.com>
Date: Thu, 9 May 2013 11:56:24 -0700
Subject: [PATCH 23/37] netdev: Add pktgen hooks to dev.c rx logic.

Signed-off-by: Ben Greear <greearb@candelatech.com>
---
 net/core/dev.c | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/net/core/dev.c b/net/core/dev.c
index 0b8b991..ff3ba72 100644
--- a/net/core/dev.c
+++ b/net/core/dev.c
@@ -154,6 +154,23 @@
 /* This should be increased if a protocol with a bigger head is added. */
 #define GRO_MAX_HEAD (MAX_HEADER + 128)
 
+#if defined(CONFIG_NET_PKTGEN) || defined(CONFIG_NET_PKTGEN_MODULE)
+#include "pktgen.h"
+
+#warning "Compiling dev.c for pktgen.";
+
+int (*handle_pktgen_hook)(struct sk_buff *skb) = NULL;
+EXPORT_SYMBOL(handle_pktgen_hook);
+
+static __inline__ int handle_pktgen_rcv(struct sk_buff* skb) {
+	if (handle_pktgen_hook) {
+		return handle_pktgen_hook(skb);
+	}
+	return -1;
+}
+#endif
+
+
 static DEFINE_SPINLOCK(ptype_lock);
 static DEFINE_SPINLOCK(offload_lock);
 struct list_head ptype_base[PTYPE_HASH_SIZE] __read_mostly;
@@ -4476,6 +4493,17 @@ static int __netif_receive_skb_core(struct sk_buff *skb, bool pfmemalloc)
 		skb->vlan_tci = 0;
 	}
 
+#if defined(CONFIG_NET_PKTGEN) || defined(CONFIG_NET_PKTGEN_MODULE)
+	if ((skb->dev->pkt_dev) &&
+	    (handle_pktgen_rcv(skb) >= 0)) {
+		/* Pktgen may consume the packet, no need to send
+		 * to further protocols.
+		 */
+		ret = NET_RX_SUCCESS;
+		goto out;
+	}
+#endif
+
 	type = skb->protocol;
 
 	/* deliver only exact match when indicated */
-- 
2.4.11

